code/game/objects/items/devices/radio/radio.dm

```diff
@@ -105,16 +105,18 @@
 /obj/item/radio/AltClick(mob/user)
 	if(headset)
 		. = ..()
-	else
+	else if(user.canUseTopic(src, !issilicon(user), TRUE, FALSE))
```

Comment:
shouldn't you also be checking canUseTopic if headset?

Reply:
From my understanding, headsets have different behavior, and the `headset` var is used to bypass the standard radio behavior. This means for headsets we just do nothing and call parent proc, because headsets already executed their own special code. This way, for headsets this code acts like the proc wasn't defined, passing through directly.

Now, if the issue is present in headsets, and we care about it, this check should be additionally defined on the headset subtype. I haven't checked it, mostly because it seems really unimportant and I didn't want to stretch the scope of the PR too much.